h2o-algos/src/main/java/hex/adaboost/AdaBoostModel.java

```diff
@@ -0,0 +1,134 @@
+package hex.adaboost;
+
+import hex.Model;
+import hex.ModelCategory;
+import hex.ModelMetrics;
+import hex.ModelMetricsBinomial;
+import hex.tree.drf.DRFModel;
+import org.apache.log4j.Logger;
+import water.*;
+
+public class AdaBoostModel extends Model<AdaBoostModel, AdaBoostModel.AdaBoostParameters, AdaBoostModel.AdaBoostOutput> {
+    private static final Logger LOG = Logger.getLogger(AdaBoostModel.class);
+
+    public enum Algorithm {DRF, GLM, GBM, AUTO}
+
+    public AdaBoostModel(Key<AdaBoostModel> selfKey, AdaBoostParameters parms,
+                         AdaBoostOutput output) {
+        super(selfKey, parms, output);
+    }
+
+    @Override
+    public ModelMetrics.MetricBuilder makeMetricBuilder(String[] domain) {
+        if (_output.getModelCategory() == ModelCategory.Binomial) {
+            return new ModelMetricsBinomial.MetricBuilderBinomial(domain);
+        }
+        throw H2O.unimpl("AdaBoost currently support only binary classification");
+    }
+
+    @Override
+    protected String[] makeScoringNames(){
+        return new String[]{"predict", "p0", "p1"};
+    }
+    
+    @Override
+    protected double[] score0(double[] data, double[] preds) {
+        double alphas0 = 0;
+        double alphas1 = 0;
+        double linearCombination = 0;
+        for (int i = 0; i < _output.alphas.length; i++) {
+            Model model = DKV.getGet(_output.models[i]);
+            if (model.score(data) == 0) {
+                linearCombination += _output.alphas[i]*-1;
+                alphas0 += _output.alphas[i];
+            } else {
+                linearCombination += _output.alphas[i]*1;
+                alphas1 += _output.alphas[i];
```

Comment:
For score = 1, we can just have _output.alphas[i] and no need for _output.alphas[i]*1.

Reply:
yes, thanks